### ISSUE TITLE
Bug >2.0.0 - \r only newlines are ignored

### ISSUE THREAD

**Issue by matthew-dean** [less/less.js#2370]
I had a bug show up in a Less library I've been working on after I upgraded Less. I was able to back-check all versions to confirm the code works in 1.7.5 and breaks in 2.0.0-2.2.0.

I've isolated the test case to this code:

in ```core.less```
```
@import "core2";
#lib.run();
```
in ```core2.less```
```
//
#lib {
    .run() { }
}
```
Error message: ```NameError: #lib.run is undefined```

Without the comment, it compiles fine.

If I move the comment here:
```
#lib {
    //
    .run() { }
}
```
The error is: ```ParseError: Unrecognised input. Possibly missing something```.

Neither error occurs in 1.7.5.

***EDIT: To help debug this, I should also note that if I move the code from core2.less to core.less, it works. And if I remove the namespace, I get the same errors. So it seems to be a combination of ```@import``` + comment + mixin.***

**Comment by lukeapage** [comment]
Can you upload the actual files? I've copied both examples into files with the same filename and I can't reproduce either problem.
I'm using lessc.. is that how you are testing or.. ?

**Comment by matthew-dean** [comment]
Oh, interesting. They both should be saved as UTF-8. Yeah, I was testing with lessc (npm install -g less@ver) to make sure there wasn't an abnormality in my integration that was causing it. I got the errors to happen in both my original files and my new reduced test case files. Let me see if I can attach them here.

**Comment by matthew-dean** [comment]
Okay, I've added both files (core.less and core2.less) to a tests folder here: https://github.com/Crunch/Granola/tree/master/tests

**Comment by lukeapage** [comment]
I havent yet downloaded, but look at

https://github.com/Crunch/Granola/blob/master/tests/
<https://github.com/Crunch/Granola/blob/master/tests/core2.less>core2
<https://github.com/Crunch/Granola/blob/master/tests/core2.less>.less
<https://github.com/Crunch/Granola/blob/master/tests/core2.less>

Notice how the github viewer only labels one line, where as core 1 has
every line labelled. I need to look in a hex viewer. Could it be you only
have \r and no \n and less used to cope with it.. ?

**Comment by matthew-dean** [comment]
Hmm... it appears you may be correct. I see that too. ...And just to confirm, I removed both files, recreated both, and entered the lines from scratch in Sublime. After that, it compiles. Yay!

So it seems fixable when that occurs, but just the same, would it be trivial to get Less to recognize both as line endings? \r with optional \n ? Especially since it was supported in the past? It was a head-banger because the error insisted the problem was an undefined mixin, and not a missing invisible unicode character. What do you think?

**Comment by matthew-dean** [comment]
I seemed to recall that Mac text files often historically encoded just one character, so did a quick search. See: http://www.bluesock.org/~willg/dev/ascii.html

Even though Github treats it as one line, Sublime on Mac does not, for example. It looks like the correct way to detect line endings is to match either \r or \n or \n\r or \r\n, so I would say if Less is not doing that anymore, that's a genuine line-ending detection bug.

(FYI: I opened my file in a hex editor. My line endings were \r, which is consistent with Mac text files.)
